### DISCUSSION TITLE
libreVNA Communication with MATLAB

### DISCUSSION THREAD

**Discussion by Frankshah007** [jankae/LibreVNA#217] [General]
Hello Everyone,
I wanted to connect my libreVNA with MATLAB. As it doesnot create any COM port, I am having difficulties to coonect with MATLAB. Can anyone give me a code for MATLAB to handshake with libreVNA ? Any ideas is highly appreciated.

Thank You Everyone

**Comment by jankae** [comment]
There is no direct support for Matlab. The LibreVNA uses a custom USB protocol, which explains why it does not show up as a COM port. And even if it did, you would need to implement the protocol between the LibreVNA and the LibreVNA-GUI. It is documented [here](https://github.com/jankae/LibreVNA/blob/master/Documentation/DeveloperInfo/Device_protocol_v13.pdf), but I don't recommend you try that route (you would need to implement all the data parsing and VNA calibration routines yourself, since these are handled on the PC-side).

Instead, I suggest you look into controlling the LibreVNA-GUI via Matlab (it supports SCPI over a TCP-Server, I am sure Matlab can work with that). The documentation for the SCPI API is [here](https://github.com/jankae/LibreVNA/blob/master/Documentation/UserManual/ProgrammingGuide.pdf).

**Reply by Frankshah007** [reply]
Thank you very much for your promt response. Can you share any code for MATLAB by which i can read data from GUI ?

**Reply by jankae** [reply]
No, I don't have any Matlab code, sorry. Please refer to the Matlab documentation on SCPI commands:
https://de.mathworks.com/products/instrument/supported/scpi.html
This, together with the SCPI API of the LibreVNA-GUI, should give you control of the LibreVNA from Matlab. But you will have to implement the Matlab side yourself.

**Reply by PrabhavManchanda** [reply]
@jankae I have a similar issue, the LibreVNA is not responding to *IDN?. It goes on a timeout when I use Matlab to communicate with it.

**Reply by jankae** [reply]
Can you please try the provided SCPI examples? If these are working, the problem must be on the Matlab side

**Reply by PrabhavManchanda** [reply]
% Create a TCPIP object to connect to the SCPI server
vnaIP = 'localhost';  % Change this to the IP address of your SCPI server
vnaPort = 19542;      % Change this to the port number of your SCPI server
vna = tcpip(vnaIP, vnaPort);

% Open the connection
fopen(vna);


response = fscanf(vna,'*IDN?');
disp(response);
fclose(vna);

**Reply by jankae** [reply]
No, I meant the actual example I provided: https://github.com/jankae/LibreVNA/blob/master/Documentation/UserManual/SCPI_Examples/simple_generator_control.py

Please try that first. If you are not getting any response with that as well, you don't even need to attempt it with Matlab. But if it works, something must be going wrong with your Matlab code.

**Reply by PrabhavManchanda** [reply]
I have never used python, I get the following error on pycharm:

Traceback (most recent call last):
  File "C:\Users\manchanda\PycharmProject\equations\main.py", line 4, in <module>
    from libreVNA import libreVNA
ModuleNotFoundError: No module named 'libreVNA'

**Reply by jankae** [reply]
You need the libreVNA.py file in the same folder

**Reply by PrabhavManchanda** [reply]
Still gives me the same error

**Reply by PrabhavManchanda** [reply]
Yeah, but I edited the code and used the libreVNA code inside the simple_generator, it works now. The GUI is also showing external influence of the code.
Thank you for the help
